### PR TITLE
Feature/parse assembly design

### PR DESCRIPTION
**Pull request by n-scholes**

### CODE REVIEW

File: antha/AnthaStandardLibrary/Packages/Parser/xlsx_parser.go [hunk 1]
```diff
@@ -118,3 +147,26 @@ func ParseExcel(filename string) ([]enzymes.Assemblyparameters, error) {
...
+	}
+
+	printer := func(s string) {
+		_, _ = f.WriteString(s)
+	}
+	err = generateCSVFromXLSXsheetBinary(data, sheetIndex, printer)
+	return
+}
+
+func ParseExcelBinary(data []byte) ([]enzymes.Assemblyparameters, error) {
```
Comment (chr1sgrant) [inline antha/AnthaStandardLibrary/Packages/Parser/xlsx_parser.go]: Please add comment for Public function

File: antha/AnthaStandardLibrary/Packages/Parser/gdxparser.go [hunk 2]
```diff
@@ -96,6 +96,13 @@ func ParseGDX(filename string) ([]enzymes.Assemblyparameters, error) {
 	}
 }
 
+func ParseGDXBinary(data []byte) ([]enzymes.Assemblyparameters, error) {
```
Comment (chr1sgrant) [inline antha/AnthaStandardLibrary/Packages/Parser/gdxparser.go]: Please add comment for Public function

File: antha/AnthaStandardLibrary/Packages/Parser/gdxparser.go [hunk 3]
```diff
@@ -96,6 +96,17 @@ func ParseGDX(filename string) ([]enzymes.Assemblyparameters, error) {
 	}
 }
 
+// Parses the contents of a typeIIs assembly design file in gdx format.
+// The output will be []Assembly Parameters which can be used in the Assemblysimulator or Digestionsimulator functions.
```
Comment (chr1sgrant) [inline antha/AnthaStandardLibrary/Packages/Parser/gdxparser.go]: enzymes.Assemblysimulator() and enzymes.Digestionsimulator() functions

Comment (chr1sgrant) [inline antha/AnthaStandardLibrary/Packages/Parser/gdxparser.go]: []enzymes.AssemblyParameters

Comment (chr1sgrant) [inline antha/AnthaStandardLibrary/Packages/Parser/gdxparser.go]: same for xlsx comment

File: antha/AnthaStandardLibrary/Packages/Parser/gdxparser.go [hunk 4]
```diff
@@ -96,6 +96,10 @@ func ParseGDX(filename string) ([]enzymes.Assemblyparameters, error) {
 	}
 }
 
+// Parses the contents of a typeIIs assembly design file in gdx format.
```
Comment (chr1sgrant) [inline antha/AnthaStandardLibrary/Packages/Parser/gdxparser.go]: should mention the example uploaded file here

File: antha/AnthaStandardLibrary/Packages/Parser/xlsx_parser.go [hunk 5]
```diff
@@ -161,6 +161,10 @@ func xlsxparserBinary(data []byte, sheetIndex int, outputprefix string) (f *os.F
 	return
 }
 
+// Parses the contents of a typeIIs assembly design file in xlsx format.
```
Comment (chr1sgrant) [inline antha/AnthaStandardLibrary/Packages/Parser/xlsx_parser.go]: should mention the example uploaded file here

File: antha/AnthaStandardLibrary/Packages/Parser/gdxparser.go [hunk 6]
```diff
@@ -96,8 +96,8 @@ func ParseGDX(filename string) ([]enzymes.Assemblyparameters, error) {
 	}
 }
 
-// Parses the contents of a typeIIs assembly design file in gdx format.
-// The output will be []Assembly Parameters which can be used in the Assemblysimulator or Digestionsimulator functions.
+// Parses the contents of a typeIIs assembly design file in gdx format. An example file is provided: "Assembly_Input_Controls.xlsx"
```
Comment (chr1sgrant) [inline antha/AnthaStandardLibrary/Packages/Parser/gdxparser.go]: "Assembly_Input_Controls.gdx"

**Comment by chr1sgrant** [comment]
@synthacebot test this please

**Comment by ddn0** [comment]
@synthacebot test this please

**Pull request merged**